### PR TITLE
feat(android): java sample

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

java sample

## Linear Issue Reference

LOG-1092

## Testing

java-sample

### CODE REVIEW

File: android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    private final MutableLiveData<IdTokenClaims> _idTokenClaims = new MutableLiveData<>();
+    public LiveData<IdTokenClaims> getIdTokenClaims() {
+        return _idTokenClaims;
+    }
+
+    private final MutableLiveData<LogtoException> _logtoException = new MutableLiveData<>();
+    public LiveData<LogtoException> getLogtoException() {
+        return _logtoException;
+    }
```
Comment (gao-sun) [inline android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java:49]: what's these for?

Comment (xiaoyijun) [reply android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java:49]: [Android ViewModel](https://developer.android.com/topic/libraries/architecture/viewmodel?gclid=Cj0KCQiAmKiQBhClARIsAKtSj-k1oJZwWN7baeAtovZ4BJzuUOdYT80F7-mXX_eMbkM-leNSlcUtDqkaAqAUEALw_wcB&gclsrc=aw.ds)

File: android-sample-java/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                android:exported="true">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+
+    </application>
+
+</manifest>
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply android-sample-java/app/src/main/AndroidManifest.xml:27]: It's fine to have a new line, although it's auto-generated.
Updated.

File: android-sample-java/app/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    implementation("androidx.appcompat:appcompat:1.3.1")
+    implementation("androidx.core:core:1.6.0")
+    implementation("com.google.code.gson:gson:2.8.9")
+
+    implementation("androidx.legacy:legacy-support-v4:1.0.0")
+    implementation("com.google.android.material:material:1.4.0")
+    implementation("androidx.constraintlayout:constraintlayout:2.1.2")
+    implementation("androidx.navigation:navigation-fragment:2.3.5")
+    implementation("androidx.navigation:navigation-ui:2.3.5")
+}
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply android-sample-java/app/build.gradle.kts:33]: Updated.

File: android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+public class LogtoViewModel extends AndroidViewModel {
+
+    public LogtoViewModel(@NonNull Application application) {
+        super(application);
+    }
+
+    private LogtoConfig logtoConfig = new LogtoConfig(
+            "https://logto.dev",
+            "z4skkM1Z8LLVSl1JCmVZO",
```
Comment (wangsijie) [inline android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java:23]: Do we need a database seed issue to ensure the clientId is valid on user's self-hosted Logto core?

**Review by wangsijie — APPROVED** [review]

File: android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java (hunk 4 above)
Comment (gao-sun) [reply android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java:23]: it's more like a doc issue? let's figure out it later

File: android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java (hunk 1 above)
Comment (gao-sun) [reply android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java:49]: looks a little bit weird. would you like to add an issue for a deeper dive on this? fine to keep as it is now

**Review by gao-sun — APPROVED** [review]

File: android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java (hunk 4 above)
Comment (xiaoyijun) [reply android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java:23]: LOG-1554

File: android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java (hunk 1 above)
Comment (xiaoyijun) [reply android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java:49]: LOG-1555

**Pull request merged**